### ISSUE TITLE
CLIJ_blur3DFast give wrong output

### ISSUE THREAD

**Issue by bpavie** [clij/clij#16]
I tried to run the following macro:

```javascript
// Get test data
run("T1 Head (2.4M, 16-bits)");
input = getTitle();
getDimensions(width, height, channels, slices, frames);

// define under which name the result should be saved
blurred = "Blurred";

// Init GPU
run("CLIJ Macro Extensions", "cl_device=NVS 5200M");
Ext.CLIJ_clear();

// push images to GPU
Ext.CLIJ_push(input);

// cleanup ImageJ
run("Close All");

// Blur in GPU
//Ext.CLIJ_mean3DBox(input, blurred, 3, 3, 3);
Ext.CLIJ_blur3DFast(input, blurred, 3, 3, 2);

// Get results back from GPU
Ext.CLIJ_pull(blurred);

// Cleanup by the end
Ext.CLIJ_clear();
```

But the output of it looks wrong:
![image](https://user-images.githubusercontent.com/1775952/51239997-1a094e00-197b-11e9-829f-ed0082b3e15f.png)

Result is the same whether I fill `"cl_device=NVS 5200M"` or leave it blank (`"cl_device="`)

I I use the regular 3D blur (`Ext.CLIJ_blur3D(input, blurred, 3, 3, 2);`) it works:
![image](https://user-images.githubusercontent.com/1775952/51240326-bf242680-197b-11e9-8efe-263e724d5eee.png)

**Comment by haesleinhuepf** [comment]
Hey @bpavie ,

thanks a lot for this bug report. I really appreciate that! 

Could you please check, which other OpenCL-Devices are available in your laptop? You can do so in almost any CLIJ menu dialog:
![image](https://user-images.githubusercontent.com/12660498/51316358-b5222680-1a54-11e9-9232-6f8e5751b025.png)

I could imagine that it works if you select an "Intel HD" GPU as cl_device. If your laptop has one. 

The issue is that the NVS 5200M is a bit old and I'm not sure if I can find a similar system to reproduce the issue. Or are you by chance coming to the NEUBIAS conference and bring that laptop? :-)

Thanks!

Cheers,
Robert

**Comment by haesleinhuepf** [comment]
Hey @bpavie ,

thanks a lot for this bug report. I really appreciate that!

Could you please check, which other OpenCL-Devices are available in your laptop? You can do so in almost any CLIJ menu dialog:
![image](https://user-images.githubusercontent.com/12660498/51316547-4e513d00-1a55-11e9-9053-903f2d96a031.png)

I could imagine that it works if you select an "Intel HD" GPU as cl_device. If your laptop has one.

The issue is that the NVS 5200M is a bit old and I'm not sure if I can find a similar system to reproduce the issue. Or are you by chance coming to the NEUBIAS conference and bring that laptop? :-)

Thanks!

Cheers,
Robert

**Comment by bpavie** [comment]
Hi Robert,

Yes, it wi working the the intel **Intel(R) HD Graphics 4000**, so if I switch to` run("CLIJ Macro Extensions", "cl_device=Intel(R) HD Graphics 4000");` or `run("CLIJ Macro Extensions", "cl_device=");` it works, but not with the NVIDIA card.
I will be at NEUBIAS so we can see together ;-)
Benjamin

**Comment by haesleinhuepf** [comment]
Hi Benjamin,

ah great. Seeing CLIJ running on this "old" Intel GPU is also great! Let's have a closer look at your NVidia GPU at NEUBIAS. Looking forward to meet you again! :-)

Cheers,
Robert

**Comment by chalkie666** [comment]
@bpavie does it work if you set the image pixel type to 32 bit?

**Comment by haesleinhuepf** [comment]
Hey @bpavie,

I just got my hands on a GPU with similar issues. It turned out, older NVidia cards only support OpenCL 1.1 and don't support 3D images 🙄 . 

I improved OpenCL 1.1 support in the recent [clij 1.0.5-BETA pre-release](https://github.com/clij/clij/releases/tag/1.0.5-BETA). With this, the bug should be fixed on your computer. Would you be so kind and download the clij-core.jar and clij_.jar, replace the older versions in the `jars` and `plugins` directory of your Fiji installation and try if the bug is fixed? Also please note that compared to your example code above the method `blur3DFast()` was renamed to `blur3D()` in the meantime.

Thanks!

Cheers,
Robert

**Comment by haesleinhuepf** [comment]
Update: The mentioned bugfix is also available in the recently release version 1.0.7 on the [clij ImageJ update site](https://sites.imagej.net/clij/) and on [github](https://github.com/clij/clij/releases/tag/1.0.7). :-)
